visualization/include/pcl/visualization/impl/pcl_visualizer.hpp

```diff
@@ -1878,8 +1903,36 @@ pcl::visualization::PCLVisualizer::updatePolygonMesh (
 
   // Update the cells
   cells = vtkSmartPointer<vtkCellArray>::New ();
-  vtkIdType *cell = cells->WritePointer (verts.size (), verts.size () * (max_size_of_polygon + 1));
   int idx = 0;
+
+#ifdef VTK_CELL_ARRAY_V2
```

Comment:
If this is a repeat, perhaps extract this into a function in details namespace outside the class. Event the cpp seems to use something like this